### PR TITLE
Remove obsolete versions as of foreman 2.3

### PR DESCRIPTION
**Pull request by pcreech**

### CODE REVIEW

File: koji/collection-mash-split.py
Comment (ekohl) [inline koji/collection-mash-split.py:394]: 2.1 is not yet obsolete since 2.3 is not yet GA.

**Review by ekohl — CHANGES_REQUESTED** [review]
As it's written now, this is only valid when 2.3 goes GA. That's a few weeks ahead so I'll leave it up to you to split it into 2 PRs or wait until then.

**Comment by pcreech** [comment]
Since 2.0 has been around for a while, i'm fine leaving this open as is until foreman 2.3 goes GA

File: koji/collection-mash-split.py
Comment (evgeni) [inline koji/collection-mash-split.py:419]: this needs a rebase ;)

Comment (pcreech) [reply koji/collection-mash-split.py:419]: rebased on latest master

File: koji/collection-mash-split.py
Comment (ekohl) [inline koji/collection-mash-split.py:432]: There is still a last 3.16.2 planned (https://community.theforeman.org/t/release-meeting-notes-2020-12-02/21509).

**Review by ekohl — COMMENTED** [review]
Blocked on Katello 3.16.2

File: koji/collection-mash-split.py
Comment (pcreech) [reply koji/collection-mash-split.py:432]: now that 3.16.2 is released, are we still blocked on this?

Comment (ekohl) [reply koji/collection-mash-split.py:432]: I don't think so. @ianballou can you confirm that there will be no more 3.16 releases?

File: koji/collection-mash-split.py
Comment (evgeni) [inline koji/collection-mash-split.py:397]: instead of `.append`, you now could just add it in the real `mashes` definition the line above?

**Review by evgeni — CHANGES_REQUESTED** [review]
the following files still have references to 2.1 and/or katello ≤ 3.16:
* puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/testKatello.groovy
* puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/vars/katello/3.16.groovy
* puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_deb_dependency.yaml
* puppet/modules/jenkins_job_builder/files/centos.org/jobs/katello-pipelines.yml
* puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/pipeline/katello-rpm-pipeline.yaml

I'd be happy if these are also cleaned up once we have the OK that Katello 3.16 is EOL (and 3.18 is officially released)

**Comment by ianballou** [comment]
I have some bugs set for a 3.16.3, but I'll move them up. We should be good here.

**Comment by evgeni** [comment]
needs rebase after #1526 got merged.

**Comment by pcreech** [comment]
@evgeni @ekohl Updated, apologies for the delay

**Review by evgeni — APPROVED** [review]

**Pull request merged**
